### DISCUSSION TITLE
Create errors from User::SaveOperation, then render errors in ECR template,    Hash(Symbol, Array(String))#each auto-splatting feature not work

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1703] [Q&A] [answered]
I use avram with kemal, following is a example.

e.g. i create a errors like this:

```
errors # => {:password_confirmation => ["must match"]}
```

When i check it type use typeof, it is `Hash(Symbol, Array(String))`.

It should work like this, because auto-splatting, right?

```cr
x = {:x => [100, 200], :y => [300, 400]}

x.each do |name, errors|
  p name, errors.join(", ")
end
```

```
:x
"100, 200"
:y
"300, 400"
```

Then, i try to render it in ECR template, like this:

```erb
<div style="color: red">
  <% errors.each do |name, errors| %>
    <%= name %> <%= errors.join(", ") %> 
  <% end %>
</div>
```

It raise error like this:

```erb
 30 | <% errors.each do |name, errors| %>
                     ^
Error: too many block parameters (given 2, expected maximum 1)
```

Why?

thank you.

**Comment by asterite** [comment]
Guess: `errors` is not what you think it is.

Try outputting the value of errros type. Something like: `<%= typeof(errors) %>` and tell us what you get.

Also, note that in the first snippet you have something called `x` but in the ECR you have `errors`.

**Comment by zw963** [comment]
Thank you. @asterite 

Okay, i add p! debug in both ECR template and cursor block, please check screenshot. 

![image](https://user-images.githubusercontent.com/549126/175816362-1f7565ae-5b75-48ee-bc08-722fb2fd9264.png)

Following is log output:

typeof(errors) # => Hash(Symbol, Array(String))
errors # => {:password_confirmation => ["must match"]}

Following is ECR output:

Hash(Symbol, Array(String)) {:password_confirmation => ["must match"]} 

Both of them is same exactly.

---------------


> Also, note that in the first snippet you have something called `x` but in the ECR you have `errors`.

Okay, i create a new example for this.

```cr
errors = {:password_confirmation => ["error1"]}
p! typeof(errors)

errors.each do |name, errors|
  p name, errors
end
```

It works, following is result:

```cr
typeof(errors) # => Hash(Symbol, Array(String))
:password_confirmation
["error1"]
```

Why ECR not work?

```ecr
<div style="color: red">
  <% errors.each do |name, errors| %>
    <%= name %> 
  <% end %>
</div>
```

**Reply by asterite** [reply]
I just tried it. It works for me.

```crystal
# foo.cr
require "ecr"

errors = {:password_confirmation => ["error1"]}
puts ECR.render("greeter.ecr")
```

This is "greeter.ecr":

```
<div style="color: red">
  <% errors.each do |name, errors| %>
    <%= name %>
  <% end %>
</div>
```

Run `crystal foo.cr`

**Reply by Blacksmoke16** [reply]
Try and reproduce it w/o using these extra libs. I just tried:

```cr
require "ecr"

errors = {:password_confirmation => ["error1"]}

# test.ecr

# <div style="color: red">
#   <% errors.each do |name, errors| %>
#     <%= name %> <%= errors.join(", ") %> 
#   <% end %>
# </div>
puts ECR.render "#{__DIR__}/../test.ecr"
# <div style="color: red">
#     password_confirmation error1
# </div>
```
and it works perfectly fine?

**Comment by zw963** [comment, marked as answer]
Oops! , it my fault, because i have code like `errors = [] of String` in another kemal block which share same ECR view file and pass different type variable into it.

low level error!
